### ISSUE TITLE
[feat] Expose GetLastError (win32) 

### ISSUE THREAD

**Issue by xan105** [Koromix/koffi#112]
Similar to #25 ; Would it be possible to expose [GetLastError()](https://learn.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror) under Windows ?

Access it via FFI is not really a solution as Koffi own syscall will change the result (_my understanding of the problem_).

```js
import koffi from 'koffi';

const lib = koffi.load('kernel32.dll');
const setLastError = lib.stdcall('SetLastError', 'void', [koffi.types.long]);
const getLastError = lib.stdcall('GetLastError', koffi.types.long, []);

setLastError(100);
const errno = getLastError();
console.log(errno); 
/*always 126 (?!) [ 'The specified module could not be found', 'ERROR_MOD_NOT_FOUND' ]

for reference the equivalent code in node-ffi always return 0;
node-ffi-napi returns the expected 100 but 0 when using .async()
*/
```

Googling around it seems other FFI implementations suffer from the same problem.

It would be rather helpful to have GetLastError() when dealing with the win32api.
`Koffi.errno()` could perhaps return the result from GetLastError() under Windows ?

**Comment by Koromix** [comment]
When replicating this, I found that it was very inconsistent; the value would sometimes remain, but not always. It's probably caused by the V8 engine or Node.js.

I've added some code to preserve the last Win32 error value, and published Koffi [2.6.10](https://koffi.dev/changelog#koffi-2-6-10-2023-11-29).

With this version your code should work as excpected.

**Comment by xan105** [comment]
Code does indeed work as expected with 2.6.10 ! Even with asynchronous call; Amazing👏.
Thank you 😃.
